e2e/pages/manageCases/createCase/C100/miamPolicyUpgrade/miamPolicyUpgrade3Page.ts

```diff
@@ -0,0 +1,195 @@
+  provideEvidenceNoInput = "#mpuNoDomesticAbuseEvidenceReason",
+  uploadFileInput = "#mpuDomesticAbuseEvidenceDocument_0_domesticAbuseDocument",
+}
+
+export class MiamPolicyUpgrade3Page {
+  public static async miamPolicyUpgrade3Page({
+    page: page,
+    accessibilityTest: accessibilityTest,
+    errorMessaging: errorMessaging,
+    yesNoMiamPolicyUpgrade: yesNoMiamPolicyUpgrade,
+  }: MiamPolicyUpgrade3PageOptions): Promise<void> {
+    await this.checkPageLoads({
+      page: page,
+      accessibilityTest: accessibilityTest,
+    });
+    if (errorMessaging) {
+      await this.triggerErrorMessages(page);
+    }
+    await this.fillInFields({
+      page: page,
+      yesNoMiamPolicyUpgrade: yesNoMiamPolicyUpgrade,
+    });
+  }
+
+  private static async checkPageLoads({
+    page: page,
+    accessibilityTest: accessibilityTest,
+  }: checkPageLoadsOptions): Promise<void> {
+    await page.waitForSelector(
+      `${Selectors.h2}:text-is("${MiamPolicyUpgrade3Content.h2}")`,
+    );
+    await Promise.all([
+      Helpers.checkVisibleAndPresent(
+        page,
+        `${Selectors.GovukHeadingL}:text-is("${MiamPolicyUpgrade3Content.pageTitle}")`,
+        1,
+      ),
+      Helpers.checkVisibleAndPresent(
+        page,
+        `${Selectors.p}:text-is("${MiamPolicyUpgrade3Content.p}")`,
+        1,
+      ),
+      Helpers.checkVisibleAndPresent(
+        page,
+        `${Selectors.GovukFormHint}:text-is("${MiamPolicyUpgrade3Content.formHint}")`,
+        1,
+      ),
+      Helpers.checkVisibleAndPresent(
+        page,
+        `${Selectors.GovukFormLabel}:text-is("${MiamPolicyUpgrade3Content.formLabelYes}")`,
+        1,
+      ),
+      Helpers.checkVisibleAndPresent(
+        page,
+        `${Selectors.GovukFormLabel}:text-is("${MiamPolicyUpgrade3Content.formLabelNo}")`,
+        1,
+      ),
+      Helpers.checkGroup(
+        page,
+        2,
+        MiamPolicyUpgrade3Content,
+        "formLabelQuestion",
+        `${Selectors.GovukFormLabel}`,
+      ),
+      Helpers.checkGroup(
+        page,
+        22,
+        MiamPolicyUpgrade3Content,
+        "formLabel",
+        `${Selectors.GovukFormLabel}`,
+      ),
+    ]);
+    if (accessibilityTest) {
+      await AccessibilityTestHelper.run(page);
+    }
+  }
+
+  public static async triggerErrorMessages(page: Page): Promise<void> {
+    await page.click(`${UniqueSelectors.provideEvidenceYes}`);
+    await this.provideEvidence_yes(page);
+    const fileInput = page.locator(`${UniqueSelectors.uploadFileInput}`);
+    await fileInput.setInputFiles(config.testOdtFile);
+    await page.waitForTimeout(5000);
+    await Helpers.checkVisibleAndPresent(
+      page,
+      `${Selectors.GovukErrorMessage}:text-is("${MiamPolicyUpgrade3Content.errorMessageFileUpload}")`,
+      1,
+    );
+  }
+
+  private static async fillInFields({
+    page: page,
+    yesNoMiamPolicyUpgrade: yesNoMiamPolicyUpgrade,
+  }: fillinFieldsOptions): Promise<void> {
+    for (let i = 1; i <= 22; i++) {
```

Comment:
This is fine for now, but in future you would have a radio enum that is separate for this, and you would do a for loop thats 
`(for radio in radioButtons) {`